internal/elasticsearch/ingest/processors_test.go

```diff
@@ -162,6 +198,90 @@ processors:
 			content: []byte(`foo123"`),
 			wantErr: true,
 		},
+		{
+			name:   "Yaml single line processor",
+			format: "yml",
+			content: []byte(`---
+processors:
+  - set: { field: "event.category", value: "web" }`),
+			expected: []Processor{
+				{Type: "set", FirstLine: 3, LastLine: 3},
+			},
+		},
+		{
+			name:   "Yaml multiline processor",
+			format: "yml",
+			content: []byte(`---
+processors:
+  - script:
+      source: |
+        def a = 1;
+        def b = 2;
+`),
+			expected: []Processor{
+				{Type: "script", FirstLine: 3, LastLine: 6},
+			},
+		},
+		{
+			name:   "Yaml script with empty line characters",
+			format: "yml",
+			content: []byte(`---
+processors:
+  - script:
+      description: Drops null/empty values recursively.
+      tag: script_drop_null_empty_values
+      lang: painless
+      source: "def a = b \n; def b = 2; \n"
+`),
+			expected: []Processor{
+				{Type: "script", FirstLine: 3, LastLine: 7},
+			},
```

Comment:
Updating this test example with two processors, I think it does not report the expected lines in the coverage.

For instance if this test is modified to include two processors whose definition is multiline like this

```
			content: []byte(`---
processors:
  - script:
      description: Drops null/empty values recursively.
      tag: script_drop_null_empty_values
      lang: painless
      source: "def a = b \n; def b = 2; \n"
  - script:
      lang: painless
      source: "def a = b \n; def b = 2; \n def c = 4; \n def d = 5; \n"
  - set:
      field: "foo"
      value: "bar"
      
      
`),
```

This gives this result:
```
[]ingest.Processor{ingest.Processor{Type:"script", FirstLine:3, LastLine:9}, ingest.Processor{Type:"script", FirstLine:8, LastLine:13}, ingest.Processor{Type:"set", FirstLine:11, LastLine:13}}
```


But according to that:
- the second processors starts (line 8) before the first processor finishes (line 9)
- the second processor finishes in the same line as the third processor.

Should it be the result this ? 
```
```
[]ingest.Processor{ingest.Processor{Type:"script", FirstLine:3, LastLine:7}, ingest.Processor{Type:"script", FirstLine:8, LastLine:10}, ingest.Processor{Type:"script", FirstLine:11, LastLine:13}}
```

I think to keep the coverage reports reporting the right lines, if that scenario happens (a processors is defined as oneline with  some `\n` in it), that should be reported as just one line. Since in the YAML file, it is in the same line too. If that is not the case, probably the coverage reports would set incorrect lines as covered or not covered.

Reply:
I need to revise the logic for determining the end of a processor. Simply counting line breaks doesn't work well, as in cases like:
```
- grok:
    tag: Extract header
    field: message
    trim_value: " \n\n"
```
will count, for example, 5 lines instead of 4.

One other option is to consider that a processor ends the line before the next one starts. However, I am running into issues when setting the LastLine number for the last processor. I can't just set it to the end of the file because there could be other elements afterward, like:

```
- rename:
    field: source.as.organization_name
    target_field: source.as.organization.name
    ignore_missing: true
on_failure:
- set:
    field: error.message
    value: '{{ _ingest.on_failure_message }}'
```

Reply:
The problem is, when using `yaml.Unmarshal`, the YAML parser will interpret
`"a = 1\nb = 2"`
as
```
a = 1
b = 2
```